### PR TITLE
be able to set event revision number

### PR DESCRIPTION
**Pull request by unixslayer**

### CODE REVIEW

File: packages/PdoEventSourcing/src/Prooph/EcotoneEventStoreProophWrapper.php [hunk 1]
```diff
@@ -137,7 +137,12 @@ private function convertToEcotoneEvents(Iterator $streamEvents, bool $deserializ
             $events[] = Event::createWithType(
                 $eventName,
                 $deserialize ? $this->conversionService->convert($event->payload(), $sourcePHPType, $PHPMediaType, $eventName, $PHPMediaType) : $event->payload(),
-                $event->metadata()
+                array_merge(
+                    [
+                        MessageHeaders::REVISION => 1,
```
Comment (dgafka) [inline packages/PdoEventSourcing/src/Prooph/EcotoneEventStoreProophWrapper.php:142]: Is this for backward compatibility?

File: packages/PdoEventSourcing/src/EventSourcingRepository.php [hunk 2]
```diff
@@ -121,6 +122,7 @@ public function save(array $identifiers, string $aggregateClassName, array $even
                     $this->headerMapper->mapFromMessageHeaders($metadata),
                     [
                         MessageHeaders::MESSAGE_ID => Uuid::uuid4()->toString(),
+                        MessageHeaders::REVISION => $this->getRevision($event),
```
Comment (dgafka) [inline packages/PdoEventSourcing/src/EventSourcingRepository.php:125]: How can this be used for deserialization?

File: packages/PdoEventSourcing/src/Prooph/EcotoneEventStoreProophWrapper.php (hunk 1 above)
Comment (unixslayer) [reply packages/PdoEventSourcing/src/Prooph/EcotoneEventStoreProophWrapper.php:142]: yes

File: packages/PdoEventSourcing/src/EventSourcingRepository.php (hunk 2 above)
Comment (unixslayer) [reply packages/PdoEventSourcing/src/EventSourcingRepository.php:125]: one step at a time. revision can still be used in debugging. converters do not have access to metadata but I was thinking to either let them or create actual upcaster

Comment (dgafka) [reply packages/PdoEventSourcing/src/EventSourcingRepository.php:125]: Ok my friend, merged :)

**Comment by dgafka** [comment]
Released with `1.50.0`, thanks :)

**Pull request merged**